plugins/org.jkiss.dbeaver.ui.editors.connection/src/org/jkiss/dbeaver/ui/dialogs/driver/EditMavenArtifactDialog.java

```diff
@@ -19,107 +19,201 @@
         super(shell);
-        this.library = library == null ?
-            new DriverLibraryMavenArtifact(driver, DBPDriverLibrary.FileType.jar, "", MavenArtifactReference.VERSION_PATTERN_RELEASE) : library;
-    }
-
-    public DriverLibraryMavenArtifact getLibrary() {
-        return library;
+        this.driver = driver;
+        this.originalArtifact = library;
     }
 
     @Override
-    protected boolean isResizable()
-    {
+    protected boolean isResizable() {
         return true;
     }
 
+    @NotNull
+    public List<DriverLibraryMavenArtifact> getArtifacts() {
+        return artifacts;
+    }
+
     @Override
-    protected Control createDialogArea(Composite parent)
-    {
+    protected Control createDialogArea(Composite parent) {
         getShell().setText(UIConnectionMessages.dialog_edit_driver_edit_maven_title);
-
         Composite composite = (Composite) super.createDialogArea(parent);
-        ((GridLayout)composite.getLayout()).numColumns = 2;
+        ((GridLayout) composite.getLayout()).numColumns = 1;
+        {
+            TabFolder tabFolder = new TabFolder(composite, SWT.TOP | SWT.FLAT);
+            GridData gridData = new GridData(GridData.FILL_BOTH);
+            tabFolder.setLayoutData(gridData);
+            if (originalArtifact == null) {
+                createRawTab(tabFolder);
+            }
+            createManualTab(tabFolder);
+            tabFolder.addSelectionListener(new SelectionAdapter() {
+                @Override
+                public void widgetSelected(SelectionEvent e) {
+                    if (tabFolder.getSelectionIndex() == 1) {
+                        artifacts.clear();
+                        manualSelected = true;
+                    } else {
+                        manualSelected = false;
+                    }
+                }
+            });
+        }
+        {
+            Group settingsGroup = UIUtils.createControlGroup(composite, UIConnectionMessages.dialog_edit_driver_edit_maven_settings, 1, GridData.FILL_HORIZONTAL, 0);
+
+            Button ignoreDependenciesCheckbox = UIUtils.createCheckbox(settingsGroup, "Ignore transient dependencies", "Do not include library dependencies", originalArtifact != null && originalArtifact.isIgnoreDependencies(), 2);
+            ignoreDependenciesCheckbox.addSelectionListener(new SelectionAdapter() {
+                @Override
+                public void widgetSelected(SelectionEvent e) {
+                    ignoreDependencies = ignoreDependenciesCheckbox.getSelection();
+                }
+            });
+
+            Button loadOptionalDependenciesCheckbox = UIUtils.createCheckbox(settingsGroup, "Load optional dependencies", "Include all optional dependencies", originalArtifact != null && originalArtifact.isIgnoreDependencies(), 2);
+            ignoreDependenciesCheckbox.addSelectionListener(new SelectionAdapter() {
```

Comment:
A typo. Should be `loadOptionalDependenciesCheckbox` instead of `ignoreDependenciesCheckbox`.